[jvm-packages] Xgboost4spark 1.1.1 broken and consistently does not work

Some models’ predictions fail on the following error:
Check failed: weights_.Size() == num_row_ (15363 vs. 15362) : Size of weights must equal to number of rows.
The numbers in the error are of course not always the same (but the subtraction is always 1).
The same data/model works on xgboost 0.9.

<details><summary>Full error log</summary>
<p>

```
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by org.apache.spark.unsafe.Platform (file:/opt/spark-3.0.0-bin-hadoop2.7/jars/spark-unsafe_2.12-3.0.0.jar) to constructor java.nio.DirectByteBuffer(long,int)
WARNING: Please consider reporting this to the maintainers of org.apache.spark.unsafe.Platform
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
20/07/22 21:18:37 WARN NativeCodeLoader: Unable to load native-hadoop library for your platform... using builtin-java classes where applicable
log4j:WARN No appenders could be found for logger (org.apache.hadoop.hive.conf.HiveConf).
log4j:WARN Please initialize the log4j system properly.
log4j:WARN See http://logging.apache.org/log4j/1.2/faq.html#noconfig for more info.
Using Spark's default log4j profile: org/apache/spark/log4j-defaults.properties
20/07/22 21:18:38 INFO SparkContext: Running Spark version 3.0.0
20/07/22 21:18:38 INFO ResourceUtils: ==============================================================
20/07/22 21:18:38 INFO ResourceUtils: Resources for spark.driver:

20/07/22 21:18:38 INFO ResourceUtils: ==============================================================
20/07/22 21:18:38 INFO SparkContext: Submitted application: ml.dmlc.xgboost4j.scala.example.spark.Foobar
20/07/22 21:18:38 INFO SecurityManager: Changing view acls to: ubuntu
20/07/22 21:18:38 INFO SecurityManager: Changing modify acls to: ubuntu
20/07/22 21:18:38 INFO SecurityManager: Changing view acls groups to: 
20/07/22 21:18:38 INFO SecurityManager: Changing modify acls groups to: 
20/07/22 21:18:38 INFO SecurityManager: SecurityManager: authentication disabled; ui acls disabled; users  with view permissions: Set(ubuntu); groups with view permissions: Set(); users  with modify permissions: Set(ubuntu); groups with modify permissions: Set()
20/07/22 21:18:38 INFO Utils: Successfully started service 'sparkDriver' on port 43551.
20/07/22 21:18:38 INFO SparkEnv: Registering MapOutputTracker
20/07/22 21:18:38 INFO SparkEnv: Registering BlockManagerMaster
20/07/22 21:18:38 INFO BlockManagerMasterEndpoint: Using org.apache.spark.storage.DefaultTopologyMapper for getting topology information
20/07/22 21:18:38 INFO BlockManagerMasterEndpoint: BlockManagerMasterEndpoint up
20/07/22 21:18:38 INFO SparkEnv: Registering BlockManagerMasterHeartbeat
20/07/22 21:18:38 INFO DiskBlockManager: Created local directory at /tmp/blockmgr-b42fb227-2e6f-4e72-b801-8857a693bfea
20/07/22 21:18:38 INFO MemoryStore: MemoryStore started with capacity 434.4 MiB
20/07/22 21:18:38 INFO SparkEnv: Registering OutputCommitCoordinator
20/07/22 21:18:38 INFO Utils: Successfully started service 'SparkUI' on port 4040.
20/07/22 21:18:38 INFO SparkUI: Bound SparkUI to 0.0.0.0, and started at http://ip-172-31-56-3.us-west-2.compute.internal:4040
20/07/22 21:18:38 INFO SparkContext: Added JAR file:/home/ubuntu/xgboost/jvm-packages/xgboost4j-tester/./target/xgboost4j-tester_2.12-1.0-SNAPSHOT-jar-with-dependencies.jar at spark://ip-172-31-56-3.us-west-2.compute.internal:43551/jars/xgboost4j-tester_2.12-1.0-SNAPSHOT-jar-with-dependencies.jar with timestamp 1595452718827
20/07/22 21:18:38 INFO Executor: Starting executor ID driver on host ip-172-31-56-3.us-west-2.compute.internal
20/07/22 21:18:38 INFO Utils: Successfully started service 'org.apache.spark.network.netty.NettyBlockTransferService' on port 35247.
20/07/22 21:18:38 INFO NettyBlockTransferService: Server created on ip-172-31-56-3.us-west-2.compute.internal:35247
20/07/22 21:18:38 INFO BlockManager: Using org.apache.spark.storage.RandomBlockReplicationPolicy for block replication policy
20/07/22 21:18:38 INFO BlockManagerMaster: Registering BlockManager BlockManagerId(driver, ip-172-31-56-3.us-west-2.compute.internal, 35247, None)
20/07/22 21:18:38 INFO BlockManagerMasterEndpoint: Registering block manager ip-172-31-56-3.us-west-2.compute.internal:35247 with 434.4 MiB RAM, BlockManagerId(driver, ip-172-31-56-3.us-west-2.compute.internal, 35247, None)
20/07/22 21:18:38 INFO BlockManagerMaster: Registered BlockManager BlockManagerId(driver, ip-172-31-56-3.us-west-2.compute.internal, 35247, None)
20/07/22 21:18:38 INFO BlockManager: Initialized BlockManager: BlockManagerId(driver, ip-172-31-56-3.us-west-2.compute.internal, 35247, None)
20/07/22 21:18:39 INFO SharedState: Setting hive.metastore.warehouse.dir ('null') to the value of spark.sql.warehouse.dir ('file:/home/ubuntu/xgboost/jvm-packages/xgboost4j-tester/spark-warehouse').
20/07/22 21:18:39 INFO SharedState: Warehouse path is 'file:/home/ubuntu/xgboost/jvm-packages/xgboost4j-tester/spark-warehouse'.
20/07/22 21:18:39 INFO InMemoryFileIndex: It took 29 ms to list leaf files for 1 paths.
20/07/22 21:18:40 INFO SparkContext: Starting job: parquet at Foobar.scala:26
20/07/22 21:18:40 INFO DAGScheduler: Got job 0 (parquet at Foobar.scala:26) with 1 output partitions
20/07/22 21:18:40 INFO DAGScheduler: Final stage: ResultStage 0 (parquet at Foobar.scala:26)
20/07/22 21:18:40 INFO DAGScheduler: Parents of final stage: List()
20/07/22 21:18:40 INFO DAGScheduler: Missing parents: List()
20/07/22 21:18:40 INFO DAGScheduler: Submitting ResultStage 0 (MapPartitionsRDD[1] at parquet at Foobar.scala:26), which has no missing parents
20/07/22 21:18:40 INFO MemoryStore: Block broadcast_0 stored as values in memory (estimated size 73.2 KiB, free 434.3 MiB)
20/07/22 21:18:40 INFO MemoryStore: Block broadcast_0_piece0 stored as bytes in memory (estimated size 26.2 KiB, free 434.3 MiB)
20/07/22 21:18:40 INFO BlockManagerInfo: Added broadcast_0_piece0 in memory on ip-172-31-56-3.us-west-2.compute.internal:35247 (size: 26.2 KiB, free: 434.4 MiB)
20/07/22 21:18:40 INFO SparkContext: Created broadcast 0 from broadcast at DAGScheduler.scala:1200
20/07/22 21:18:40 INFO DAGScheduler: Submitting 1 missing tasks from ResultStage 0 (MapPartitionsRDD[1] at parquet at Foobar.scala:26) (first 15 tasks are for partitions Vector(0))
20/07/22 21:18:40 INFO TaskSchedulerImpl: Adding task set 0.0 with 1 tasks
20/07/22 21:18:40 INFO TaskSetManager: Starting task 0.0 in stage 0.0 (TID 0, ip-172-31-56-3.us-west-2.compute.internal, executor driver, partition 0, PROCESS_LOCAL, 7538 bytes)
20/07/22 21:18:40 INFO Executor: Running task 0.0 in stage 0.0 (TID 0)
20/07/22 21:18:40 INFO Executor: Fetching spark://ip-172-31-56-3.us-west-2.compute.internal:43551/jars/xgboost4j-tester_2.12-1.0-SNAPSHOT-jar-with-dependencies.jar with timestamp 1595452718827
20/07/22 21:18:40 INFO TransportClientFactory: Successfully created connection to ip-172-31-56-3.us-west-2.compute.internal/172.31.56.3:43551 after 19 ms (0 ms spent in bootstraps)
20/07/22 21:18:40 INFO Utils: Fetching spark://ip-172-31-56-3.us-west-2.compute.internal:43551/jars/xgboost4j-tester_2.12-1.0-SNAPSHOT-jar-with-dependencies.jar to /tmp/spark-64acb100-e2c0-44d9-a5f4-547437ae4630/userFiles-76e88cdf-c18e-4fc9-8f1d-53353ffc68f2/fetchFileTemp12665909510458585974.tmp
20/07/22 21:18:40 INFO Executor: Adding file:/tmp/spark-64acb100-e2c0-44d9-a5f4-547437ae4630/userFiles-76e88cdf-c18e-4fc9-8f1d-53353ffc68f2/xgboost4j-tester_2.12-1.0-SNAPSHOT-jar-with-dependencies.jar to class loader
20/07/22 21:18:40 INFO Executor: Finished task 0.0 in stage 0.0 (TID 0). 51176 bytes result sent to driver
20/07/22 21:18:40 INFO TaskSetManager: Finished task 0.0 in stage 0.0 (TID 0) in 633 ms on ip-172-31-56-3.us-west-2.compute.internal (executor driver) (1/1)
20/07/22 21:18:40 INFO TaskSchedulerImpl: Removed TaskSet 0.0, whose tasks have all completed, from pool 
20/07/22 21:18:40 INFO DAGScheduler: ResultStage 0 (parquet at Foobar.scala:26) finished in 0.744 s
20/07/22 21:18:40 INFO DAGScheduler: Job 0 is finished. Cancelling potential speculative or zombie tasks for this job
20/07/22 21:18:40 INFO TaskSchedulerImpl: Killing all running tasks in stage 0: Stage finished
20/07/22 21:18:40 INFO DAGScheduler: Job 0 finished: parquet at Foobar.scala:26, took 0.776912 s
20/07/22 21:18:41 INFO MemoryStore: Block broadcast_1 stored as values in memory (estimated size 127.5 KiB, free 434.2 MiB)
20/07/22 21:18:41 INFO MemoryStore: Block broadcast_1_piece0 stored as bytes in memory (estimated size 23.6 KiB, free 434.2 MiB)
20/07/22 21:18:41 INFO BlockManagerInfo: Added broadcast_1_piece0 in memory on ip-172-31-56-3.us-west-2.compute.internal:35247 (size: 23.6 KiB, free: 434.4 MiB)
20/07/22 21:18:41 INFO SparkContext: Created broadcast 1 from textFile at ReadWrite.scala:587
20/07/22 21:18:41 INFO FileInputFormat: Total input paths to process : 1
20/07/22 21:18:41 INFO SparkContext: Starting job: first at ReadWrite.scala:587
20/07/22 21:18:41 INFO DAGScheduler: Got job 1 (first at ReadWrite.scala:587) with 1 output partitions
20/07/22 21:18:41 INFO DAGScheduler: Final stage: ResultStage 1 (first at ReadWrite.scala:587)
20/07/22 21:18:41 INFO DAGScheduler: Parents of final stage: List()
20/07/22 21:18:41 INFO DAGScheduler: Missing parents: List()
20/07/22 21:18:41 INFO DAGScheduler: Submitting ResultStage 1 (/home/ubuntu/model/metadata MapPartitionsRDD[3] at textFile at ReadWrite.scala:587), which has no missing parents
20/07/22 21:18:41 INFO MemoryStore: Block broadcast_2 stored as values in memory (estimated size 4.1 KiB, free 434.2 MiB)
20/07/22 21:18:41 INFO MemoryStore: Block broadcast_2_piece0 stored as bytes in memory (estimated size 2.4 KiB, free 434.1 MiB)
20/07/22 21:18:41 INFO BlockManagerInfo: Added broadcast_2_piece0 in memory on ip-172-31-56-3.us-west-2.compute.internal:35247 (size: 2.4 KiB, free: 434.3 MiB)
20/07/22 21:18:41 INFO SparkContext: Created broadcast 2 from broadcast at DAGScheduler.scala:1200
20/07/22 21:18:41 INFO DAGScheduler: Submitting 1 missing tasks from ResultStage 1 (/home/ubuntu/model/metadata MapPartitionsRDD[3] at textFile at ReadWrite.scala:587) (first 15 tasks are for partitions Vector(0))
20/07/22 21:18:41 INFO TaskSchedulerImpl: Adding task set 1.0 with 1 tasks
20/07/22 21:18:41 INFO TaskSetManager: Starting task 0.0 in stage 1.0 (TID 1, ip-172-31-56-3.us-west-2.compute.internal, executor driver, partition 0, PROCESS_LOCAL, 7384 bytes)
20/07/22 21:18:41 INFO Executor: Running task 0.0 in stage 1.0 (TID 1)
20/07/22 21:18:41 INFO HadoopRDD: Input split: file:/home/ubuntu/model/metadata/part-00000:0+210
20/07/22 21:18:41 INFO Executor: Finished task 0.0 in stage 1.0 (TID 1). 1138 bytes result sent to driver
20/07/22 21:18:41 INFO TaskSetManager: Finished task 0.0 in stage 1.0 (TID 1) in 34 ms on ip-172-31-56-3.us-west-2.compute.internal (executor driver) (1/1)
20/07/22 21:18:41 INFO TaskSchedulerImpl: Removed TaskSet 1.0, whose tasks have all completed, from pool 
20/07/22 21:18:41 INFO DAGScheduler: ResultStage 1 (first at ReadWrite.scala:587) finished in 0.041 s
20/07/22 21:18:41 INFO DAGScheduler: Job 1 is finished. Cancelling potential speculative or zombie tasks for this job
20/07/22 21:18:41 INFO TaskSchedulerImpl: Killing all running tasks in stage 1: Stage finished
20/07/22 21:18:41 INFO DAGScheduler: Job 1 finished: first at ReadWrite.scala:587, took 0.045394 s
20/07/22 21:18:41 INFO MemoryStore: Block broadcast_3 stored as values in memory (estimated size 127.5 KiB, free 434.0 MiB)
20/07/22 21:18:41 INFO MemoryStore: Block broadcast_3_piece0 stored as bytes in memory (estimated size 23.6 KiB, free 434.0 MiB)
20/07/22 21:18:41 INFO BlockManagerInfo: Added broadcast_3_piece0 in memory on ip-172-31-56-3.us-west-2.compute.internal:35247 (size: 23.6 KiB, free: 434.3 MiB)
20/07/22 21:18:41 INFO SparkContext: Created broadcast 3 from textFile at ReadWrite.scala:587
20/07/22 21:18:41 INFO FileInputFormat: Total input paths to process : 1
20/07/22 21:18:41 INFO SparkContext: Starting job: first at ReadWrite.scala:587
20/07/22 21:18:41 INFO DAGScheduler: Got job 2 (first at ReadWrite.scala:587) with 1 output partitions
20/07/22 21:18:41 INFO DAGScheduler: Final stage: ResultStage 2 (first at ReadWrite.scala:587)
20/07/22 21:18:41 INFO DAGScheduler: Parents of final stage: List()
20/07/22 21:18:41 INFO DAGScheduler: Missing parents: List()
20/07/22 21:18:41 INFO DAGScheduler: Submitting ResultStage 2 (/home/ubuntu/model/stages/0_XGBoostRegressor_dfcc0f11d073/metadata MapPartitionsRDD[5] at textFile at ReadWrite.scala:587), which has no missing parents
20/07/22 21:18:41 INFO MemoryStore: Block broadcast_4 stored as values in memory (estimated size 4.2 KiB, free 434.0 MiB)
20/07/22 21:18:41 INFO MemoryStore: Block broadcast_4_piece0 stored as bytes in memory (estimated size 2.4 KiB, free 434.0 MiB)
20/07/22 21:18:41 INFO BlockManagerInfo: Added broadcast_4_piece0 in memory on ip-172-31-56-3.us-west-2.compute.internal:35247 (size: 2.4 KiB, free: 434.3 MiB)
20/07/22 21:18:41 INFO SparkContext: Created broadcast 4 from broadcast at DAGScheduler.scala:1200
20/07/22 21:18:41 INFO DAGScheduler: Submitting 1 missing tasks from ResultStage 2 (/home/ubuntu/model/stages/0_XGBoostRegressor_dfcc0f11d073/metadata MapPartitionsRDD[5] at textFile at ReadWrite.scala:587) (first 15 tasks are for partitions Vector(0))
20/07/22 21:18:41 INFO TaskSchedulerImpl: Adding task set 2.0 with 1 tasks
20/07/22 21:18:41 INFO TaskSetManager: Starting task 0.0 in stage 2.0 (TID 2, ip-172-31-56-3.us-west-2.compute.internal, executor driver, partition 0, PROCESS_LOCAL, 7423 bytes)
20/07/22 21:18:41 INFO Executor: Running task 0.0 in stage 2.0 (TID 2)
20/07/22 21:18:41 INFO HadoopRDD: Input split: file:/home/ubuntu/model/stages/0_XGBoostRegressor_dfcc0f11d073/metadata/part-00000:0+1093
20/07/22 21:18:41 INFO Executor: Finished task 0.0 in stage 2.0 (TID 2). 2023 bytes result sent to driver
20/07/22 21:18:41 INFO TaskSetManager: Finished task 0.0 in stage 2.0 (TID 2) in 12 ms on ip-172-31-56-3.us-west-2.compute.internal (executor driver) (1/1)
20/07/22 21:18:41 INFO TaskSchedulerImpl: Removed TaskSet 2.0, whose tasks have all completed, from pool 
20/07/22 21:18:41 INFO DAGScheduler: ResultStage 2 (first at ReadWrite.scala:587) finished in 0.018 s
20/07/22 21:18:41 INFO DAGScheduler: Job 2 is finished. Cancelling potential speculative or zombie tasks for this job
20/07/22 21:18:41 INFO TaskSchedulerImpl: Killing all running tasks in stage 2: Stage finished
20/07/22 21:18:41 INFO DAGScheduler: Job 2 finished: first at ReadWrite.scala:587, took 0.021352 s
20/07/22 21:18:42 INFO MemoryStore: Block broadcast_5 stored as values in memory (estimated size 127.5 KiB, free 433.9 MiB)
20/07/22 21:18:42 INFO MemoryStore: Block broadcast_5_piece0 stored as bytes in memory (estimated size 23.6 KiB, free 433.8 MiB)
20/07/22 21:18:42 INFO BlockManagerInfo: Added broadcast_5_piece0 in memory on ip-172-31-56-3.us-west-2.compute.internal:35247 (size: 23.6 KiB, free: 434.3 MiB)
20/07/22 21:18:42 INFO SparkContext: Created broadcast 5 from textFile at DefaultXGBoostParamsReader.scala:82
20/07/22 21:18:42 INFO FileInputFormat: Total input paths to process : 1
20/07/22 21:18:42 INFO SparkContext: Starting job: first at DefaultXGBoostParamsReader.scala:82
20/07/22 21:18:42 INFO DAGScheduler: Got job 3 (first at DefaultXGBoostParamsReader.scala:82) with 1 output partitions
20/07/22 21:18:42 INFO DAGScheduler: Final stage: ResultStage 3 (first at DefaultXGBoostParamsReader.scala:82)
20/07/22 21:18:42 INFO DAGScheduler: Parents of final stage: List()
20/07/22 21:18:42 INFO DAGScheduler: Missing parents: List()
20/07/22 21:18:42 INFO DAGScheduler: Submitting ResultStage 3 (/home/ubuntu/model/stages/0_XGBoostRegressor_dfcc0f11d073/metadata MapPartitionsRDD[7] at textFile at DefaultXGBoostParamsReader.scala:82), which has no missing parents
20/07/22 21:18:42 INFO MemoryStore: Block broadcast_6 stored as values in memory (estimated size 4.2 KiB, free 433.8 MiB)
20/07/22 21:18:42 INFO MemoryStore: Block broadcast_6_piece0 stored as bytes in memory (estimated size 2.4 KiB, free 433.8 MiB)
20/07/22 21:18:42 INFO BlockManagerInfo: Added broadcast_6_piece0 in memory on ip-172-31-56-3.us-west-2.compute.internal:35247 (size: 2.4 KiB, free: 434.3 MiB)
20/07/22 21:18:42 INFO SparkContext: Created broadcast 6 from broadcast at DAGScheduler.scala:1200
20/07/22 21:18:42 INFO DAGScheduler: Submitting 1 missing tasks from ResultStage 3 (/home/ubuntu/model/stages/0_XGBoostRegressor_dfcc0f11d073/metadata MapPartitionsRDD[7] at textFile at DefaultXGBoostParamsReader.scala:82) (first 15 tasks are for partitions Vector(0))
20/07/22 21:18:42 INFO TaskSchedulerImpl: Adding task set 3.0 with 1 tasks
20/07/22 21:18:42 INFO TaskSetManager: Starting task 0.0 in stage 3.0 (TID 3, ip-172-31-56-3.us-west-2.compute.internal, executor driver, partition 0, PROCESS_LOCAL, 7423 bytes)
20/07/22 21:18:42 INFO Executor: Running task 0.0 in stage 3.0 (TID 3)
20/07/22 21:18:42 INFO HadoopRDD: Input split: file:/home/ubuntu/model/stages/0_XGBoostRegressor_dfcc0f11d073/metadata/part-00000:0+1093
20/07/22 21:18:42 INFO Executor: Finished task 0.0 in stage 3.0 (TID 3). 2023 bytes result sent to driver
20/07/22 21:18:42 INFO TaskSetManager: Finished task 0.0 in stage 3.0 (TID 3) in 10 ms on ip-172-31-56-3.us-west-2.compute.internal (executor driver) (1/1)
20/07/22 21:18:42 INFO TaskSchedulerImpl: Removed TaskSet 3.0, whose tasks have all completed, from pool 
20/07/22 21:18:42 INFO DAGScheduler: ResultStage 3 (first at DefaultXGBoostParamsReader.scala:82) finished in 0.019 s
20/07/22 21:18:42 INFO DAGScheduler: Job 3 is finished. Cancelling potential speculative or zombie tasks for this job
20/07/22 21:18:42 INFO TaskSchedulerImpl: Killing all running tasks in stage 3: Stage finished
20/07/22 21:18:42 INFO DAGScheduler: Job 3 finished: first at DefaultXGBoostParamsReader.scala:82, took 0.022431 s
[21:18:42] [XGBoost C API invocation] int XGBoosterCreate(void* const*, xgboost::bst_ulong, void**)()
[21:18:42] [XGBoost C API invocation] int XGBoosterLoadModelFromBuffer(BoosterHandle, const void*, xgboost::bst_ulong)()
[21:18:42] WARNING: /home/ubuntu/xgboost/src/learner.cc:736: Loading model from XGBoost < 1.0.0, consider saving it again for improved compatibility
20/07/22 21:18:42 INFO Instrumentation: [cb23d225] training finished
20/07/22 21:18:42 INFO Instrumentation: [4c6f0f45] training finished
20/07/22 21:18:42 INFO MemoryStore: Block broadcast_7 stored as values in memory (estimated size 64.0 B, free 433.8 MiB)
[21:18:42] [XGBoost C API invocation] int XGBoosterGetModelRaw(BoosterHandle, xgboost::bst_ulong*, const char**)()
20/07/22 21:18:42 INFO MemoryStore: Block broadcast_7_piece0 stored as bytes in memory (estimated size 25.5 KiB, free 433.8 MiB)
20/07/22 21:18:42 INFO BlockManagerInfo: Added broadcast_7_piece0 in memory on ip-172-31-56-3.us-west-2.compute.internal:35247 (size: 25.5 KiB, free: 434.3 MiB)
20/07/22 21:18:42 INFO SparkContext: Created broadcast 7 from broadcast at XGBoostRegressor.scala:274
20/07/22 21:18:42 INFO BlockManagerInfo: Removed broadcast_4_piece0 on ip-172-31-56-3.us-west-2.compute.internal:35247 in memory (size: 2.4 KiB, free: 434.3 MiB)
20/07/22 21:18:42 INFO BlockManagerInfo: Removed broadcast_2_piece0 on ip-172-31-56-3.us-west-2.compute.internal:35247 in memory (size: 2.4 KiB, free: 434.3 MiB)
20/07/22 21:18:42 INFO BlockManagerInfo: Removed broadcast_3_piece0 on ip-172-31-56-3.us-west-2.compute.internal:35247 in memory (size: 23.6 KiB, free: 434.3 MiB)
20/07/22 21:18:42 INFO BlockManagerInfo: Removed broadcast_1_piece0 on ip-172-31-56-3.us-west-2.compute.internal:35247 in memory (size: 23.6 KiB, free: 434.3 MiB)
20/07/22 21:18:42 INFO BlockManagerInfo: Removed broadcast_6_piece0 on ip-172-31-56-3.us-west-2.compute.internal:35247 in memory (size: 2.4 KiB, free: 434.3 MiB)
20/07/22 21:18:42 INFO BlockManagerInfo: Removed broadcast_0_piece0 on ip-172-31-56-3.us-west-2.compute.internal:35247 in memory (size: 26.2 KiB, free: 434.4 MiB)
20/07/22 21:18:42 INFO BlockManagerInfo: Removed broadcast_5_piece0 on ip-172-31-56-3.us-west-2.compute.internal:35247 in memory (size: 23.6 KiB, free: 434.4 MiB)
20/07/22 21:18:42 INFO FileSourceStrategy: Pruning directories with: 
20/07/22 21:18:42 INFO FileSourceStrategy: Pushed Filters: 
20/07/22 21:18:42 INFO FileSourceStrategy: Post-Scan Filters: 
20/07/22 21:18:42 INFO FileSourceStrategy: Output Data Schema: struct<account_code: bigint, features_7528809678875577807: vector>
20/07/22 21:18:42 INFO MemoryStore: Block broadcast_8 stored as values in memory (estimated size 229.7 KiB, free 434.2 MiB)
20/07/22 21:18:42 INFO MemoryStore: Block broadcast_8_piece0 stored as bytes in memory (estimated size 43.6 KiB, free 434.1 MiB)
20/07/22 21:18:42 INFO BlockManagerInfo: Added broadcast_8_piece0 in memory on ip-172-31-56-3.us-west-2.compute.internal:35247 (size: 43.6 KiB, free: 434.3 MiB)
20/07/22 21:18:42 INFO SparkContext: Created broadcast 8 from rdd at XGBoostRegressor.scala:277
20/07/22 21:18:42 INFO FileSourceScanExec: Planning scan with bin packing, max size: 29262023 bytes, open cost is considered as scanning 4194304 bytes.
[21:18:42] [XGBoost C API invocation] int XGBoosterGetModelRaw(BoosterHandle, xgboost::bst_ulong*, const char**)()
20/07/22 21:18:42 INFO Instrumentation: [1adb517a] training finished
20/07/22 21:18:43 INFO CodeGenerator: Code generated in 140.54394 ms
20/07/22 21:18:43 INFO CodeGenerator: Code generated in 12.516245 ms
20/07/22 21:18:43 INFO CodeGenerator: Code generated in 28.744701 ms
20/07/22 21:18:43 INFO SparkContext: Starting job: count at Foobar.scala:30
20/07/22 21:18:43 INFO DAGScheduler: Registering RDD 20 (count at Foobar.scala:30) as input to shuffle 0
20/07/22 21:18:43 INFO DAGScheduler: Got job 4 (count at Foobar.scala:30) with 1 output partitions
20/07/22 21:18:43 INFO DAGScheduler: Final stage: ResultStage 5 (count at Foobar.scala:30)
20/07/22 21:18:43 INFO DAGScheduler: Parents of final stage: List(ShuffleMapStage 4)
20/07/22 21:18:43 INFO DAGScheduler: Missing parents: List(ShuffleMapStage 4)
20/07/22 21:18:43 INFO DAGScheduler: Submitting ShuffleMapStage 4 (MapPartitionsRDD[20] at count at Foobar.scala:30), which has no missing parents
[21:18:43] [XGBoost C API invocation] int XGBoosterGetModelRaw(BoosterHandle, xgboost::bst_ulong*, const char**)()
20/07/22 21:18:43 INFO MemoryStore: Block broadcast_9 stored as values in memory (estimated size 158.9 KiB, free 434.0 MiB)
20/07/22 21:18:43 INFO MemoryStore: Block broadcast_9_piece0 stored as bytes in memory (estimated size 59.7 KiB, free 433.9 MiB)
20/07/22 21:18:43 INFO BlockManagerInfo: Added broadcast_9_piece0 in memory on ip-172-31-56-3.us-west-2.compute.internal:35247 (size: 59.7 KiB, free: 434.3 MiB)
20/07/22 21:18:43 INFO SparkContext: Created broadcast 9 from broadcast at DAGScheduler.scala:1200
20/07/22 21:18:43 INFO DAGScheduler: Submitting 1 missing tasks from ShuffleMapStage 4 (MapPartitionsRDD[20] at count at Foobar.scala:30) (first 15 tasks are for partitions Vector(0))
20/07/22 21:18:43 INFO TaskSchedulerImpl: Adding task set 4.0 with 1 tasks
20/07/22 21:18:43 INFO TaskSetManager: Starting task 0.0 in stage 4.0 (TID 4, ip-172-31-56-3.us-west-2.compute.internal, executor driver, partition 0, PROCESS_LOCAL, 7778 bytes)
20/07/22 21:18:43 INFO Executor: Running task 0.0 in stage 4.0 (TID 4)
[21:18:43] [XGBoost C API invocation] int XGBoosterCreate(void* const*, xgboost::bst_ulong, void**)()
[21:18:43] [XGBoost C API invocation] int XGBoosterLoadModelFromBuffer(BoosterHandle, const void*, xgboost::bst_ulong)()
20/07/22 21:18:43 INFO CodeGenerator: Code generated in 19.283191 ms
20/07/22 21:18:43 INFO CodeGenerator: Code generated in 10.941417 ms
20/07/22 21:18:43 INFO FileScanRDD: Reading File path: file:///home/ubuntu/data/part-00000-3655a559-8238-46e0-a145-f4b4bbc75ad6-c000.snappy.parquet, range: 0-25067719, partition values: [empty row]
20/07/22 21:18:43 INFO InternalParquetRecordReader: RecordReader initialized will read a total of 1522672 records.
20/07/22 21:18:43 INFO InternalParquetRecordReader: at row 0. reading next block
20/07/22 21:18:43 INFO CodecPool: Got brand-new decompressor [.snappy]
20/07/22 21:18:43 INFO InternalParquetRecordReader: block read in memory in 42 ms. row count = 1522672
[21:18:44] [XGBoost C API invocation] int XGDMatrixCreateFromDataIter(void*, int (*)(DataIterHandle, int (*)(DataHolderHandle, XGBoostBatchCSR), DataHolderHandle), const char*, void**)()
[21:18:44] [XGBoost C API invocation] xgboost::data::IteratorAdapter<DataIterHandle, XGBCallbackDataIterNext, XGBoostBatchCSR>::Next()::<lambda(void*, XGBoostBatchCSR)> [with DataIterHandle = void*; XGBCallbackDataIterNext = int(void*, int (*)(void*, XGBoostBatchCSR), void*); XGBoostBatchCSR = XGBoostBatchCSR]()
[21:18:44] [XGBoost C API invocation] int XGBoosterPredict(BoosterHandle, DMatrixHandle, int, unsigned int, int, xgboost::bst_ulong*, const bst_float**)()
[21:18:44] [XGBoost C API invocation] int XGDMatrixNumRow(DMatrixHandle, xgboost::bst_ulong*)()
[21:18:44] [XGBoost C API invocation] int XGDMatrixFree(DMatrixHandle)()
20/07/22 21:18:44 INFO CodeGenerator: Code generated in 31.903108 ms
[21:18:44] [XGBoost C API invocation] int XGDMatrixCreateFromDataIter(void*, int (*)(DataIterHandle, int (*)(DataHolderHandle, XGBoostBatchCSR), DataHolderHandle), const char*, void**)()
[21:18:44] [XGBoost C API invocation] xgboost::data::IteratorAdapter<DataIterHandle, XGBCallbackDataIterNext, XGBoostBatchCSR>::Next()::<lambda(void*, XGBoostBatchCSR)> [with DataIterHandle = void*; XGBCallbackDataIterNext = int(void*, int (*)(void*, XGBoostBatchCSR), void*); XGBoostBatchCSR = XGBoostBatchCSR]()
[21:18:44] [XGBoost C API invocation] int XGBoosterPredict(BoosterHandle, DMatrixHandle, int, unsigned int, int, xgboost::bst_ulong*, const bst_float**)()
[21:18:44] [XGBoost C API invocation] int XGDMatrixNumRow(DMatrixHandle, xgboost::bst_ulong*)()
[21:18:44] [XGBoost C API invocation] int XGDMatrixFree(DMatrixHandle)()
[21:18:44] [XGBoost C API invocation] int XGDMatrixCreateFromDataIter(void*, int (*)(DataIterHandle, int (*)(DataHolderHandle, XGBoostBatchCSR), DataHolderHandle), const char*, void**)()
[21:18:44] [XGBoost C API invocation] xgboost::data::IteratorAdapter<DataIterHandle, XGBCallbackDataIterNext, XGBoostBatchCSR>::Next()::<lambda(void*, XGBoostBatchCSR)> [with DataIterHandle = void*; XGBCallbackDataIterNext = int(void*, int (*)(void*, XGBoostBatchCSR), void*); XGBoostBatchCSR = XGBoostBatchCSR]()
[21:18:44] [XGBoost C API invocation] int XGBoosterPredict(BoosterHandle, DMatrixHandle, int, unsigned int, int, xgboost::bst_ulong*, const bst_float**)()
[21:18:44] [XGBoost C API invocation] int XGDMatrixFree(DMatrixHandle)()
20/07/22 21:18:44 WARN BlockManager: Putting block rdd_16_0 failed due to exception ml.dmlc.xgboost4j.java.XGBoostError: [21:18:44] /home/ubuntu/xgboost/src/data/data.cc:524: Check failed: weights_.Size() == num_row_ (32768 vs. 32767) : Size of weights must equal to number of rows.
Stack trace:
  [bt] (0) /tmp/libxgboost4j3900874476663887786.so(dmlc::LogMessageFatal::~LogMessageFatal()+0x7c) [0x7f4b1747facc]
  [bt] (1) /tmp/libxgboost4j3900874476663887786.so(xgboost::MetaInfo::Validate(int) const+0x106) [0x7f4b174f61f6]
  [bt] (2) /tmp/libxgboost4j3900874476663887786.so(xgboost::LearnerImpl::ValidateDMatrix(xgboost::DMatrix*) const+0x37) [0x7f4b17581107]
  [bt] (3) /tmp/libxgboost4j3900874476663887786.so(xgboost::LearnerImpl::PredictRaw(xgboost::DMatrix*, xgboost::PredictionCacheEntry*, bool, unsigned int) const+0x44) [0x7f4b17581344]
  [bt] (4) /tmp/libxgboost4j3900874476663887786.so(xgboost::LearnerImpl::Predict(std::shared_ptr<xgboost::DMatrix>, bool, xgboost::HostDeviceVector<float>*, unsigned int, bool, bool, bool, bool, bool)+0x123) [0x7f4b17583663]
  [bt] (5) /tmp/libxgboost4j3900874476663887786.so(XGBoosterPredict+0x152) [0x7f4b174848d2]
  [bt] (6) /tmp/libxgboost4j3900874476663887786.so(Java_ml_dmlc_xgboost4j_java_XGBoostJNI_XGBoosterPredict+0x43) [0x7f4b17477c73]
  [bt] (7) [0x7f4c6faa1890]

.
20/07/22 21:18:44 WARN BlockManager: Block rdd_16_0 could not be removed as it was not found on disk or in memory
20/07/22 21:18:44 ERROR Executor: Exception in task 0.0 in stage 4.0 (TID 4)
ml.dmlc.xgboost4j.java.XGBoostError: [21:18:44] /home/ubuntu/xgboost/src/data/data.cc:524: Check failed: weights_.Size() == num_row_ (32768 vs. 32767) : Size of weights must equal to number of rows.
Stack trace:
  [bt] (0) /tmp/libxgboost4j3900874476663887786.so(dmlc::LogMessageFatal::~LogMessageFatal()+0x7c) [0x7f4b1747facc]
  [bt] (1) /tmp/libxgboost4j3900874476663887786.so(xgboost::MetaInfo::Validate(int) const+0x106) [0x7f4b174f61f6]
  [bt] (2) /tmp/libxgboost4j3900874476663887786.so(xgboost::LearnerImpl::ValidateDMatrix(xgboost::DMatrix*) const+0x37) [0x7f4b17581107]
  [bt] (3) /tmp/libxgboost4j3900874476663887786.so(xgboost::LearnerImpl::PredictRaw(xgboost::DMatrix*, xgboost::PredictionCacheEntry*, bool, unsigned int) const+0x44) [0x7f4b17581344]
  [bt] (4) /tmp/libxgboost4j3900874476663887786.so(xgboost::LearnerImpl::Predict(std::shared_ptr<xgboost::DMatrix>, bool, xgboost::HostDeviceVector<float>*, unsigned int, bool, bool, bool, bool, bool)+0x123) [0x7f4b17583663]
  [bt] (5) /tmp/libxgboost4j3900874476663887786.so(XGBoosterPredict+0x152) [0x7f4b174848d2]
  [bt] (6) /tmp/libxgboost4j3900874476663887786.so(Java_ml_dmlc_xgboost4j_java_XGBoostJNI_XGBoosterPredict+0x43) [0x7f4b17477c73]
  [bt] (7) [0x7f4c6faa1890]


	at ml.dmlc.xgboost4j.java.XGBoostJNI.checkCall(XGBoostJNI.java:48)
	at ml.dmlc.xgboost4j.java.Booster.predict(Booster.java:312)
	at ml.dmlc.xgboost4j.java.Booster.predict(Booster.java:381)
	at ml.dmlc.xgboost4j.scala.Booster.predict(Booster.scala:172)
	at ml.dmlc.xgboost4j.scala.spark.XGBoostRegressionModel.ml$dmlc$xgboost4j$scala$spark$XGBoostRegressionModel$$producePredictionItrs(XGBoostRegressor.scala:381)
	at ml.dmlc.xgboost4j.scala.spark.XGBoostRegressionModel$$anon$1.$anonfun$batchIterImpl$1(XGBoostRegressor.scala:310)
	at scala.collection.Iterator$$anon$11.nextCur(Iterator.scala:484)
	at scala.collection.Iterator$$anon$11.hasNext(Iterator.scala:490)
	at ml.dmlc.xgboost4j.scala.spark.XGBoostRegressionModel$$anon$1.next(XGBoostRegressor.scala:322)
	at ml.dmlc.xgboost4j.scala.spark.XGBoostRegressionModel$$anon$1.next(XGBoostRegressor.scala:278)
	at scala.collection.Iterator$$anon$10.next(Iterator.scala:459)
	at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIteratorForCodegenStage1.processNext(Unknown Source)
	at org.apache.spark.sql.execution.BufferedRowIterator.hasNext(BufferedRowIterator.java:43)
	at org.apache.spark.sql.execution.WholeStageCodegenExec$$anon$1.hasNext(WholeStageCodegenExec.scala:729)
	at org.apache.spark.sql.execution.columnar.CachedRDDBuilder$$anon$1.next(InMemoryRelation.scala:98)
	at org.apache.spark.sql.execution.columnar.CachedRDDBuilder$$anon$1.next(InMemoryRelation.scala:90)
	at org.apache.spark.storage.memory.MemoryStore.putIterator(MemoryStore.scala:222)
	at org.apache.spark.storage.memory.MemoryStore.putIteratorAsValues(MemoryStore.scala:299)
	at org.apache.spark.storage.BlockManager.$anonfun$doPutIterator$1(BlockManager.scala:1371)
	at org.apache.spark.storage.BlockManager.org$apache$spark$storage$BlockManager$$doPut(BlockManager.scala:1298)
	at org.apache.spark.storage.BlockManager.doPutIterator(BlockManager.scala:1362)
	at org.apache.spark.storage.BlockManager.getOrElseUpdate(BlockManager.scala:1186)
	at org.apache.spark.rdd.RDD.getOrCompute(RDD.scala:360)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:311)
	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:349)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:313)
	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:349)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:313)
	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:349)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:313)
	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:349)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:313)
	at org.apache.spark.shuffle.ShuffleWriteProcessor.write(ShuffleWriteProcessor.scala:59)
	at org.apache.spark.scheduler.ShuffleMapTask.runTask(ShuffleMapTask.scala:99)
	at org.apache.spark.scheduler.ShuffleMapTask.runTask(ShuffleMapTask.scala:52)
	at org.apache.spark.scheduler.Task.run(Task.scala:127)
	at org.apache.spark.executor.Executor$TaskRunner.$anonfun$run$3(Executor.scala:444)
	at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:1377)
	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:447)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
20/07/22 21:18:44 WARN TaskSetManager: Lost task 0.0 in stage 4.0 (TID 4, ip-172-31-56-3.us-west-2.compute.internal, executor driver): ml.dmlc.xgboost4j.java.XGBoostError: [21:18:44] /home/ubuntu/xgboost/src/data/data.cc:524: Check failed: weights_.Size() == num_row_ (32768 vs. 32767) : Size of weights must equal to number of rows.
Stack trace:
  [bt] (0) /tmp/libxgboost4j3900874476663887786.so(dmlc::LogMessageFatal::~LogMessageFatal()+0x7c) [0x7f4b1747facc]
  [bt] (1) /tmp/libxgboost4j3900874476663887786.so(xgboost::MetaInfo::Validate(int) const+0x106) [0x7f4b174f61f6]
  [bt] (2) /tmp/libxgboost4j3900874476663887786.so(xgboost::LearnerImpl::ValidateDMatrix(xgboost::DMatrix*) const+0x37) [0x7f4b17581107]
  [bt] (3) /tmp/libxgboost4j3900874476663887786.so(xgboost::LearnerImpl::PredictRaw(xgboost::DMatrix*, xgboost::PredictionCacheEntry*, bool, unsigned int) const+0x44) [0x7f4b17581344]
  [bt] (4) /tmp/libxgboost4j3900874476663887786.so(xgboost::LearnerImpl::Predict(std::shared_ptr<xgboost::DMatrix>, bool, xgboost::HostDeviceVector<float>*, unsigned int, bool, bool, bool, bool, bool)+0x123) [0x7f4b17583663]
  [bt] (5) /tmp/libxgboost4j3900874476663887786.so(XGBoosterPredict+0x152) [0x7f4b174848d2]
  [bt] (6) /tmp/libxgboost4j3900874476663887786.so(Java_ml_dmlc_xgboost4j_java_XGBoostJNI_XGBoosterPredict+0x43) [0x7f4b17477c73]
  [bt] (7) [0x7f4c6faa1890]


	at ml.dmlc.xgboost4j.java.XGBoostJNI.checkCall(XGBoostJNI.java:48)
	at ml.dmlc.xgboost4j.java.Booster.predict(Booster.java:312)
	at ml.dmlc.xgboost4j.java.Booster.predict(Booster.java:381)
	at ml.dmlc.xgboost4j.scala.Booster.predict(Booster.scala:172)
	at ml.dmlc.xgboost4j.scala.spark.XGBoostRegressionModel.ml$dmlc$xgboost4j$scala$spark$XGBoostRegressionModel$$producePredictionItrs(XGBoostRegressor.scala:381)
	at ml.dmlc.xgboost4j.scala.spark.XGBoostRegressionModel$$anon$1.$anonfun$batchIterImpl$1(XGBoostRegressor.scala:310)
	at scala.collection.Iterator$$anon$11.nextCur(Iterator.scala:484)
	at scala.collection.Iterator$$anon$11.hasNext(Iterator.scala:490)
	at ml.dmlc.xgboost4j.scala.spark.XGBoostRegressionModel$$anon$1.next(XGBoostRegressor.scala:322)
	at ml.dmlc.xgboost4j.scala.spark.XGBoostRegressionModel$$anon$1.next(XGBoostRegressor.scala:278)
	at scala.collection.Iterator$$anon$10.next(Iterator.scala:459)
	at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIteratorForCodegenStage1.processNext(Unknown Source)
	at org.apache.spark.sql.execution.BufferedRowIterator.hasNext(BufferedRowIterator.java:43)
	at org.apache.spark.sql.execution.WholeStageCodegenExec$$anon$1.hasNext(WholeStageCodegenExec.scala:729)
	at org.apache.spark.sql.execution.columnar.CachedRDDBuilder$$anon$1.next(InMemoryRelation.scala:98)
	at org.apache.spark.sql.execution.columnar.CachedRDDBuilder$$anon$1.next(InMemoryRelation.scala:90)
	at org.apache.spark.storage.memory.MemoryStore.putIterator(MemoryStore.scala:222)
	at org.apache.spark.storage.memory.MemoryStore.putIteratorAsValues(MemoryStore.scala:299)
	at org.apache.spark.storage.BlockManager.$anonfun$doPutIterator$1(BlockManager.scala:1371)
	at org.apache.spark.storage.BlockManager.org$apache$spark$storage$BlockManager$$doPut(BlockManager.scala:1298)
	at org.apache.spark.storage.BlockManager.doPutIterator(BlockManager.scala:1362)
	at org.apache.spark.storage.BlockManager.getOrElseUpdate(BlockManager.scala:1186)
	at org.apache.spark.rdd.RDD.getOrCompute(RDD.scala:360)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:311)
	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:349)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:313)
	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:349)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:313)
	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:349)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:313)
	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:349)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:313)
	at org.apache.spark.shuffle.ShuffleWriteProcessor.write(ShuffleWriteProcessor.scala:59)
	at org.apache.spark.scheduler.ShuffleMapTask.runTask(ShuffleMapTask.scala:99)
	at org.apache.spark.scheduler.ShuffleMapTask.runTask(ShuffleMapTask.scala:52)
	at org.apache.spark.scheduler.Task.run(Task.scala:127)
	at org.apache.spark.executor.Executor$TaskRunner.$anonfun$run$3(Executor.scala:444)
	at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:1377)
	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:447)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)

20/07/22 21:18:44 ERROR TaskSetManager: Task 0 in stage 4.0 failed 1 times; aborting job
20/07/22 21:18:44 INFO TaskSchedulerImpl: Removed TaskSet 4.0, whose tasks have all completed, from pool 
20/07/22 21:18:44 INFO TaskSchedulerImpl: Cancelling stage 4
20/07/22 21:18:44 INFO TaskSchedulerImpl: Killing all running tasks in stage 4: Stage cancelled
20/07/22 21:18:44 INFO DAGScheduler: ShuffleMapStage 4 (count at Foobar.scala:30) failed in 1.599 s due to Job aborted due to stage failure: Task 0 in stage 4.0 failed 1 times, most recent failure: Lost task 0.0 in stage 4.0 (TID 4, ip-172-31-56-3.us-west-2.compute.internal, executor driver): ml.dmlc.xgboost4j.java.XGBoostError: [21:18:44] /home/ubuntu/xgboost/src/data/data.cc:524: Check failed: weights_.Size() == num_row_ (32768 vs. 32767) : Size of weights must equal to number of rows.
Stack trace:
  [bt] (0) /tmp/libxgboost4j3900874476663887786.so(dmlc::LogMessageFatal::~LogMessageFatal()+0x7c) [0x7f4b1747facc]
  [bt] (1) /tmp/libxgboost4j3900874476663887786.so(xgboost::MetaInfo::Validate(int) const+0x106) [0x7f4b174f61f6]
  [bt] (2) /tmp/libxgboost4j3900874476663887786.so(xgboost::LearnerImpl::ValidateDMatrix(xgboost::DMatrix*) const+0x37) [0x7f4b17581107]
  [bt] (3) /tmp/libxgboost4j3900874476663887786.so(xgboost::LearnerImpl::PredictRaw(xgboost::DMatrix*, xgboost::PredictionCacheEntry*, bool, unsigned int) const+0x44) [0x7f4b17581344]
  [bt] (4) /tmp/libxgboost4j3900874476663887786.so(xgboost::LearnerImpl::Predict(std::shared_ptr<xgboost::DMatrix>, bool, xgboost::HostDeviceVector<float>*, unsigned int, bool, bool, bool, bool, bool)+0x123) [0x7f4b17583663]
  [bt] (5) /tmp/libxgboost4j3900874476663887786.so(XGBoosterPredict+0x152) [0x7f4b174848d2]
  [bt] (6) /tmp/libxgboost4j3900874476663887786.so(Java_ml_dmlc_xgboost4j_java_XGBoostJNI_XGBoosterPredict+0x43) [0x7f4b17477c73]
  [bt] (7) [0x7f4c6faa1890]


	at ml.dmlc.xgboost4j.java.XGBoostJNI.checkCall(XGBoostJNI.java:48)
	at ml.dmlc.xgboost4j.java.Booster.predict(Booster.java:312)
	at ml.dmlc.xgboost4j.java.Booster.predict(Booster.java:381)
	at ml.dmlc.xgboost4j.scala.Booster.predict(Booster.scala:172)
	at ml.dmlc.xgboost4j.scala.spark.XGBoostRegressionModel.ml$dmlc$xgboost4j$scala$spark$XGBoostRegressionModel$$producePredictionItrs(XGBoostRegressor.scala:381)
	at ml.dmlc.xgboost4j.scala.spark.XGBoostRegressionModel$$anon$1.$anonfun$batchIterImpl$1(XGBoostRegressor.scala:310)
	at scala.collection.Iterator$$anon$11.nextCur(Iterator.scala:484)
	at scala.collection.Iterator$$anon$11.hasNext(Iterator.scala:490)
	at ml.dmlc.xgboost4j.scala.spark.XGBoostRegressionModel$$anon$1.next(XGBoostRegressor.scala:322)
	at ml.dmlc.xgboost4j.scala.spark.XGBoostRegressionModel$$anon$1.next(XGBoostRegressor.scala:278)
	at scala.collection.Iterator$$anon$10.next(Iterator.scala:459)
	at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIteratorForCodegenStage1.processNext(Unknown Source)
	at org.apache.spark.sql.execution.BufferedRowIterator.hasNext(BufferedRowIterator.java:43)
	at org.apache.spark.sql.execution.WholeStageCodegenExec$$anon$1.hasNext(WholeStageCodegenExec.scala:729)
	at org.apache.spark.sql.execution.columnar.CachedRDDBuilder$$anon$1.next(InMemoryRelation.scala:98)
	at org.apache.spark.sql.execution.columnar.CachedRDDBuilder$$anon$1.next(InMemoryRelation.scala:90)
	at org.apache.spark.storage.memory.MemoryStore.putIterator(MemoryStore.scala:222)
	at org.apache.spark.storage.memory.MemoryStore.putIteratorAsValues(MemoryStore.scala:299)
	at org.apache.spark.storage.BlockManager.$anonfun$doPutIterator$1(BlockManager.scala:1371)
	at org.apache.spark.storage.BlockManager.org$apache$spark$storage$BlockManager$$doPut(BlockManager.scala:1298)
	at org.apache.spark.storage.BlockManager.doPutIterator(BlockManager.scala:1362)
	at org.apache.spark.storage.BlockManager.getOrElseUpdate(BlockManager.scala:1186)
	at org.apache.spark.rdd.RDD.getOrCompute(RDD.scala:360)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:311)
	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:349)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:313)
	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:349)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:313)
	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:349)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:313)
	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:349)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:313)
	at org.apache.spark.shuffle.ShuffleWriteProcessor.write(ShuffleWriteProcessor.scala:59)
	at org.apache.spark.scheduler.ShuffleMapTask.runTask(ShuffleMapTask.scala:99)
	at org.apache.spark.scheduler.ShuffleMapTask.runTask(ShuffleMapTask.scala:52)
	at org.apache.spark.scheduler.Task.run(Task.scala:127)
	at org.apache.spark.executor.Executor$TaskRunner.$anonfun$run$3(Executor.scala:444)
	at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:1377)
	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:447)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)

Driver stacktrace:
20/07/22 21:18:44 INFO DAGScheduler: Job 4 failed: count at Foobar.scala:30, took 1.613591 s
20/07/22 21:18:44 INFO SparkUI: Stopped Spark web UI at http://ip-172-31-56-3.us-west-2.compute.internal:4040
20/07/22 21:18:44 INFO MapOutputTrackerMasterEndpoint: MapOutputTrackerMasterEndpoint stopped!
20/07/22 21:18:44 INFO MemoryStore: MemoryStore cleared
20/07/22 21:18:44 INFO BlockManager: BlockManager stopped
20/07/22 21:18:44 INFO BlockManagerMaster: BlockManagerMaster stopped
20/07/22 21:18:44 INFO OutputCommitCoordinator$OutputCommitCoordinatorEndpoint: OutputCommitCoordinator stopped!
20/07/22 21:18:44 INFO SparkContext: Successfully stopped SparkContext
Exception in thread "main" org.apache.spark.SparkException: Job aborted due to stage failure: Task 0 in stage 4.0 failed 1 times, most recent failure: Lost task 0.0 in stage 4.0 (TID 4, ip-172-31-56-3.us-west-2.compute.internal, executor driver): ml.dmlc.xgboost4j.java.XGBoostError: [21:18:44] /home/ubuntu/xgboost/src/data/data.cc:524: Check failed: weights_.Size() == num_row_ (32768 vs. 32767) : Size of weights must equal to number of rows.
Stack trace:
  [bt] (0) /tmp/libxgboost4j3900874476663887786.so(dmlc::LogMessageFatal::~LogMessageFatal()+0x7c) [0x7f4b1747facc]
  [bt] (1) /tmp/libxgboost4j3900874476663887786.so(xgboost::MetaInfo::Validate(int) const+0x106) [0x7f4b174f61f6]
  [bt] (2) /tmp/libxgboost4j3900874476663887786.so(xgboost::LearnerImpl::ValidateDMatrix(xgboost::DMatrix*) const+0x37) [0x7f4b17581107]
  [bt] (3) /tmp/libxgboost4j3900874476663887786.so(xgboost::LearnerImpl::PredictRaw(xgboost::DMatrix*, xgboost::PredictionCacheEntry*, bool, unsigned int) const+0x44) [0x7f4b17581344]
  [bt] (4) /tmp/libxgboost4j3900874476663887786.so(xgboost::LearnerImpl::Predict(std::shared_ptr<xgboost::DMatrix>, bool, xgboost::HostDeviceVector<float>*, unsigned int, bool, bool, bool, bool, bool)+0x123) [0x7f4b17583663]
  [bt] (5) /tmp/libxgboost4j3900874476663887786.so(XGBoosterPredict+0x152) [0x7f4b174848d2]
  [bt] (6) /tmp/libxgboost4j3900874476663887786.so(Java_ml_dmlc_xgboost4j_java_XGBoostJNI_XGBoosterPredict+0x43) [0x7f4b17477c73]
  [bt] (7) [0x7f4c6faa1890]


	at ml.dmlc.xgboost4j.java.XGBoostJNI.checkCall(XGBoostJNI.java:48)
	at ml.dmlc.xgboost4j.java.Booster.predict(Booster.java:312)
	at ml.dmlc.xgboost4j.java.Booster.predict(Booster.java:381)
	at ml.dmlc.xgboost4j.scala.Booster.predict(Booster.scala:172)
	at ml.dmlc.xgboost4j.scala.spark.XGBoostRegressionModel.ml$dmlc$xgboost4j$scala$spark$XGBoostRegressionModel$$producePredictionItrs(XGBoostRegressor.scala:381)
	at ml.dmlc.xgboost4j.scala.spark.XGBoostRegressionModel$$anon$1.$anonfun$batchIterImpl$1(XGBoostRegressor.scala:310)
	at scala.collection.Iterator$$anon$11.nextCur(Iterator.scala:484)
	at scala.collection.Iterator$$anon$11.hasNext(Iterator.scala:490)
	at ml.dmlc.xgboost4j.scala.spark.XGBoostRegressionModel$$anon$1.next(XGBoostRegressor.scala:322)
	at ml.dmlc.xgboost4j.scala.spark.XGBoostRegressionModel$$anon$1.next(XGBoostRegressor.scala:278)
	at scala.collection.Iterator$$anon$10.next(Iterator.scala:459)
	at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIteratorForCodegenStage1.processNext(Unknown Source)
	at org.apache.spark.sql.execution.BufferedRowIterator.hasNext(BufferedRowIterator.java:43)
	at org.apache.spark.sql.execution.WholeStageCodegenExec$$anon$1.hasNext(WholeStageCodegenExec.scala:729)
	at org.apache.spark.sql.execution.columnar.CachedRDDBuilder$$anon$1.next(InMemoryRelation.scala:98)
	at org.apache.spark.sql.execution.columnar.CachedRDDBuilder$$anon$1.next(InMemoryRelation.scala:90)
	at org.apache.spark.storage.memory.MemoryStore.putIterator(MemoryStore.scala:222)
	at org.apache.spark.storage.memory.MemoryStore.putIteratorAsValues(MemoryStore.scala:299)
	at org.apache.spark.storage.BlockManager.$anonfun$doPutIterator$1(BlockManager.scala:1371)
	at org.apache.spark.storage.BlockManager.org$apache$spark$storage$BlockManager$$doPut(BlockManager.scala:1298)
	at org.apache.spark.storage.BlockManager.doPutIterator(BlockManager.scala:1362)
	at org.apache.spark.storage.BlockManager.getOrElseUpdate(BlockManager.scala:1186)
	at org.apache.spark.rdd.RDD.getOrCompute(RDD.scala:360)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:311)
	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:349)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:313)
	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:349)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:313)
	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:349)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:313)
	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:349)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:313)
	at org.apache.spark.shuffle.ShuffleWriteProcessor.write(ShuffleWriteProcessor.scala:59)
	at org.apache.spark.scheduler.ShuffleMapTask.runTask(ShuffleMapTask.scala:99)
	at org.apache.spark.scheduler.ShuffleMapTask.runTask(ShuffleMapTask.scala:52)
	at org.apache.spark.scheduler.Task.run(Task.scala:127)
	at org.apache.spark.executor.Executor$TaskRunner.$anonfun$run$3(Executor.scala:444)
	at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:1377)
	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:447)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)

Driver stacktrace:
	at org.apache.spark.scheduler.DAGScheduler.failJobAndIndependentStages(DAGScheduler.scala:2023)
	at org.apache.spark.scheduler.DAGScheduler.$anonfun$abortStage$2(DAGScheduler.scala:1972)
	at org.apache.spark.scheduler.DAGScheduler.$anonfun$abortStage$2$adapted(DAGScheduler.scala:1971)
	at scala.collection.mutable.ResizableArray.foreach(ResizableArray.scala:62)
	at scala.collection.mutable.ResizableArray.foreach$(ResizableArray.scala:55)
	at scala.collection.mutable.ArrayBuffer.foreach(ArrayBuffer.scala:49)
	at org.apache.spark.scheduler.DAGScheduler.abortStage(DAGScheduler.scala:1971)
	at org.apache.spark.scheduler.DAGScheduler.$anonfun$handleTaskSetFailed$1(DAGScheduler.scala:950)
	at org.apache.spark.scheduler.DAGScheduler.$anonfun$handleTaskSetFailed$1$adapted(DAGScheduler.scala:950)
	at scala.Option.foreach(Option.scala:407)
	at org.apache.spark.scheduler.DAGScheduler.handleTaskSetFailed(DAGScheduler.scala:950)
	at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.doOnReceive(DAGScheduler.scala:2203)
	at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.onReceive(DAGScheduler.scala:2152)
	at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.onReceive(DAGScheduler.scala:2141)
	at org.apache.spark.util.EventLoop$$anon$1.run(EventLoop.scala:49)
	at org.apache.spark.scheduler.DAGScheduler.runJob(DAGScheduler.scala:752)
	at org.apache.spark.SparkContext.runJob(SparkContext.scala:2093)
	at org.apache.spark.SparkContext.runJob(SparkContext.scala:2114)
	at org.apache.spark.SparkContext.runJob(SparkContext.scala:2133)
	at org.apache.spark.SparkContext.runJob(SparkContext.scala:2158)
	at org.apache.spark.rdd.RDD.$anonfun$collect$1(RDD.scala:1004)
	at org.apache.spark.rdd.RDDOperationScope$.withScope(RDDOperationScope.scala:151)
	at org.apache.spark.rdd.RDDOperationScope$.withScope(RDDOperationScope.scala:112)
	at org.apache.spark.rdd.RDD.withScope(RDD.scala:388)
	at org.apache.spark.rdd.RDD.collect(RDD.scala:1003)
	at org.apache.spark.sql.execution.SparkPlan.executeCollect(SparkPlan.scala:385)
	at org.apache.spark.sql.Dataset.$anonfun$count$1(Dataset.scala:2979)
	at org.apache.spark.sql.Dataset.$anonfun$count$1$adapted(Dataset.scala:2978)
	at org.apache.spark.sql.Dataset.$anonfun$withAction$1(Dataset.scala:3616)
	at org.apache.spark.sql.execution.SQLExecution$.$anonfun$withNewExecutionId$5(SQLExecution.scala:100)
	at org.apache.spark.sql.execution.SQLExecution$.withSQLConfPropagated(SQLExecution.scala:160)
	at org.apache.spark.sql.execution.SQLExecution$.$anonfun$withNewExecutionId$1(SQLExecution.scala:87)
	at org.apache.spark.sql.SparkSession.withActive(SparkSession.scala:763)
	at org.apache.spark.sql.execution.SQLExecution$.withNewExecutionId(SQLExecution.scala:64)
	at org.apache.spark.sql.Dataset.withAction(Dataset.scala:3614)
	at org.apache.spark.sql.Dataset.count(Dataset.scala:2978)
	at ml.dmlc.xgboost4j.scala.example.spark.Foobar$.main(Foobar.scala:30)
	at ml.dmlc.xgboost4j.scala.example.spark.Foobar.main(Foobar.scala)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.apache.spark.deploy.JavaMainApplication.start(SparkApplication.scala:52)
	at org.apache.spark.deploy.SparkSubmit.org$apache$spark$deploy$SparkSubmit$$runMain(SparkSubmit.scala:928)
	at org.apache.spark.deploy.SparkSubmit.doRunMain$1(SparkSubmit.scala:180)
	at org.apache.spark.deploy.SparkSubmit.submit(SparkSubmit.scala:203)
	at org.apache.spark.deploy.SparkSubmit.doSubmit(SparkSubmit.scala:90)
	at org.apache.spark.deploy.SparkSubmit$$anon$2.doSubmit(SparkSubmit.scala:1007)
	at org.apache.spark.deploy.SparkSubmit$.main(SparkSubmit.scala:1016)
	at org.apache.spark.deploy.SparkSubmit.main(SparkSubmit.scala)
Caused by: ml.dmlc.xgboost4j.java.XGBoostError: [21:18:44] /home/ubuntu/xgboost/src/data/data.cc:524: Check failed: weights_.Size() == num_row_ (32768 vs. 32767) : Size of weights must equal to number of rows.
Stack trace:
  [bt] (0) /tmp/libxgboost4j3900874476663887786.so(dmlc::LogMessageFatal::~LogMessageFatal()+0x7c) [0x7f4b1747facc]
  [bt] (1) /tmp/libxgboost4j3900874476663887786.so(xgboost::MetaInfo::Validate(int) const+0x106) [0x7f4b174f61f6]
  [bt] (2) /tmp/libxgboost4j3900874476663887786.so(xgboost::LearnerImpl::ValidateDMatrix(xgboost::DMatrix*) const+0x37) [0x7f4b17581107]
  [bt] (3) /tmp/libxgboost4j3900874476663887786.so(xgboost::LearnerImpl::PredictRaw(xgboost::DMatrix*, xgboost::PredictionCacheEntry*, bool, unsigned int) const+0x44) [0x7f4b17581344]
  [bt] (4) /tmp/libxgboost4j3900874476663887786.so(xgboost::LearnerImpl::Predict(std::shared_ptr<xgboost::DMatrix>, bool, xgboost::HostDeviceVector<float>*, unsigned int, bool, bool, bool, bool, bool)+0x123) [0x7f4b17583663]
  [bt] (5) /tmp/libxgboost4j3900874476663887786.so(XGBoosterPredict+0x152) [0x7f4b174848d2]
  [bt] (6) /tmp/libxgboost4j3900874476663887786.so(Java_ml_dmlc_xgboost4j_java_XGBoostJNI_XGBoosterPredict+0x43) [0x7f4b17477c73]
  [bt] (7) [0x7f4c6faa1890]


	at ml.dmlc.xgboost4j.java.XGBoostJNI.checkCall(XGBoostJNI.java:48)
	at ml.dmlc.xgboost4j.java.Booster.predict(Booster.java:312)
	at ml.dmlc.xgboost4j.java.Booster.predict(Booster.java:381)
	at ml.dmlc.xgboost4j.scala.Booster.predict(Booster.scala:172)
	at ml.dmlc.xgboost4j.scala.spark.XGBoostRegressionModel.ml$dmlc$xgboost4j$scala$spark$XGBoostRegressionModel$$producePredictionItrs(XGBoostRegressor.scala:381)
	at ml.dmlc.xgboost4j.scala.spark.XGBoostRegressionModel$$anon$1.$anonfun$batchIterImpl$1(XGBoostRegressor.scala:310)
	at scala.collection.Iterator$$anon$11.nextCur(Iterator.scala:484)
	at scala.collection.Iterator$$anon$11.hasNext(Iterator.scala:490)
	at ml.dmlc.xgboost4j.scala.spark.XGBoostRegressionModel$$anon$1.next(XGBoostRegressor.scala:322)
	at ml.dmlc.xgboost4j.scala.spark.XGBoostRegressionModel$$anon$1.next(XGBoostRegressor.scala:278)
	at scala.collection.Iterator$$anon$10.next(Iterator.scala:459)
	at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIteratorForCodegenStage1.processNext(Unknown Source)
	at org.apache.spark.sql.execution.BufferedRowIterator.hasNext(BufferedRowIterator.java:43)
	at org.apache.spark.sql.execution.WholeStageCodegenExec$$anon$1.hasNext(WholeStageCodegenExec.scala:729)
	at org.apache.spark.sql.execution.columnar.CachedRDDBuilder$$anon$1.next(InMemoryRelation.scala:98)
	at org.apache.spark.sql.execution.columnar.CachedRDDBuilder$$anon$1.next(InMemoryRelation.scala:90)
	at org.apache.spark.storage.memory.MemoryStore.putIterator(MemoryStore.scala:222)
	at org.apache.spark.storage.memory.MemoryStore.putIteratorAsValues(MemoryStore.scala:299)
	at org.apache.spark.storage.BlockManager.$anonfun$doPutIterator$1(BlockManager.scala:1371)
	at org.apache.spark.storage.BlockManager.org$apache$spark$storage$BlockManager$$doPut(BlockManager.scala:1298)
	at org.apache.spark.storage.BlockManager.doPutIterator(BlockManager.scala:1362)
	at org.apache.spark.storage.BlockManager.getOrElseUpdate(BlockManager.scala:1186)
	at org.apache.spark.rdd.RDD.getOrCompute(RDD.scala:360)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:311)
	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:349)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:313)
	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:349)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:313)
	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:349)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:313)
	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:349)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:313)
	at org.apache.spark.shuffle.ShuffleWriteProcessor.write(ShuffleWriteProcessor.scala:59)
	at org.apache.spark.scheduler.ShuffleMapTask.runTask(ShuffleMapTask.scala:99)
	at org.apache.spark.scheduler.ShuffleMapTask.runTask(ShuffleMapTask.scala:52)
	at org.apache.spark.scheduler.Task.run(Task.scala:127)
	at org.apache.spark.executor.Executor$TaskRunner.$anonfun$run$3(Executor.scala:444)
	at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:1377)
	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:447)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
20/07/22 21:18:44 INFO ShutdownHookManager: Shutdown hook called
20/07/22 21:18:45 INFO ShutdownHookManager: Deleting directory /tmp/spark-64acb100-e2c0-44d9-a5f4-547437ae4630
20/07/22 21:18:45 INFO ShutdownHookManager: Deleting directory /tmp/spark-72e7c939-d476-4b7b-885d-20c60c97eca8
```

</p>
</details>
[xgboost4j_spark_crash_log.txt](https://github.com/dmlc/xgboost/files/4962719/xgboost4j_spark_crash_log.txt)

--- a/jvm-packages/xgboost4j-spark/src/test/scala/ml/dmlc/xgboost4j/scala/spark/MissingValueHandlingSuite.scala
+++ b/jvm-packages/xgboost4j-spark/src/test/scala/ml/dmlc/xgboost4j/scala/spark/MissingValueHandlingSuite.scala
@@ -171,4 +171,31 @@ class MissingValueHandlingSuite extends FunSuite with PerTest {
     val model = new XGBoostClassifier(paramMap).fit(inputDF)
     model.transform(inputDF).collect()
   }
+
+  // https://github.com/dmlc/xgboost/pull/5929
+  test("handle the empty last row correctly with a missing value as 0") {
+    val spark = ss
+    import spark.implicits._
+    // spark uses 1.5 * (nnz + 1.0) < size as the condition to decide whether using sparse or dense
+    // vector,
+    val testDF = Seq(
+      (7.0f, 0.0f, -1.0f, 1.0f, 1.0),
+      (1.0f, 0.0f, 1.0f, 1.0f, 1.0),
+      (0.0f, 1.0f, 0.0f, 1.0f, 0.0),
+      (1.0f, 0.0f, 1.0f, 1.0f, 1.0),
+      (1.0f, -1.0f, 0.0f, 1.0f, 0.0),
+      (0.0f, 0.0f, 0.0f, 1.0f, 1.0),
+      (0.0f, 0.0f, 0.0f, 0.0f, 0.0)
+    ).toDF("col1", "col2", "col3", "col4", "label")
+    val vectorAssembler = new VectorAssembler()
+      .setInputCols(Array("col1", "col2", "col3", "col4"))
+      .setOutputCol("features")
+    val inputDF = vectorAssembler.transform(testDF).select("features", "label")
+    inputDF.show()
+    val paramMap = List("eta" -> "1", "max_depth" -> "2",
+      "objective" -> "binary:logistic", "missing" -> 0.0f,
+      "num_workers" -> 1, "allow_non_zero_for_missing" -> "true").toMap
+    val model = new XGBoostClassifier(paramMap).fit(inputDF)
+    model.transform(inputDF).collect()
+  }
 }

--- a/src/data/data.cc
+++ b/src/data/data.cc
@@ -833,9 +833,9 @@ uint64_t SparsePage::Push(const AdapterBatchT& batch, float missing, int nthread
   uint64_t max_columns = 0;
 
   // First-pass over the batch counting valid elements
-  size_t num_lines = batch.Size();
+  size_t batch_size = batch.Size();
 #pragma omp parallel for schedule(static)
-  for (omp_ulong i = 0; i < static_cast<omp_ulong>(num_lines);
+  for (omp_ulong i = 0; i < static_cast<omp_ulong>(batch_size);
        ++i) {  // NOLINT(*)
     int tid = omp_get_thread_num();
     auto line = batch.GetLine(i);
@@ -847,7 +847,7 @@ uint64_t SparsePage::Push(const AdapterBatchT& batch, float missing, int nthread
         size_t key = element.row_idx - base_rowid;
         // Adapter row index is absolute, here we want it relative to
         // current page
-        CHECK_GE(key,  builder_base_row_offset);
+        CHECK_GE(key, builder_base_row_offset);
         builder.AddBudget(key, tid);
       }
     }
@@ -856,7 +856,7 @@ uint64_t SparsePage::Push(const AdapterBatchT& batch, float missing, int nthread
 
   // Second pass over batch, placing elements in correct position
 #pragma omp parallel for schedule(static)
-  for (omp_ulong i = 0; i < static_cast<omp_ulong>(num_lines);
+  for (omp_ulong i = 0; i < static_cast<omp_ulong>(batch_size);
        ++i) {  // NOLINT(*)
     int tid = omp_get_thread_num();
     auto line = batch.GetLine(i);

--- a/src/data/simple_dmatrix.cc
+++ b/src/data/simple_dmatrix.cc
@@ -164,9 +164,12 @@ SimpleDMatrix::SimpleDMatrix(AdapterT* adapter, float missing, int nthread) {
   if (adapter->NumRows() == kAdapterUnknownSize) {
     using IteratorAdapterT
       = IteratorAdapter<DataIterHandle, XGBCallbackDataIterNext, XGBoostBatchCSR>;
+    // If AdapterT is either IteratorAdapter or FileAdapter type, use the total batch size to
+    // determine the correct number of rows, as offset_vec may be too short
     if (std::is_same<AdapterT, IteratorAdapterT>::value
         || std::is_same<AdapterT, FileAdapter>::value) {
       info_.num_row_ = total_batch_size;
+      // Ensure offset_vec.size() - 1 == [number of rows]
       while (offset_vec.size() - 1 < total_batch_size) {
         offset_vec.emplace_back(offset_vec.back());
       }

--- a/src/data/simple_dmatrix.cc
+++ b/src/data/simple_dmatrix.cc
@@ -6,6 +6,7 @@
  */
 #include <vector>
 #include <limits>
+#include <type_traits>
 #include <algorithm>
 
 #include "xgboost/data.h"
@@ -103,6 +104,8 @@ SimpleDMatrix::SimpleDMatrix(AdapterT* adapter, float missing, int nthread) {
   auto& offset_vec = sparse_page_.offset.HostVector();
   auto& data_vec = sparse_page_.data.HostVector();
   uint64_t inferred_num_columns = 0;
+  uint64_t total_batch_size = 0;
+    // batch_size is either number of rows or cols, depending on data layout
 
   adapter->BeforeFirst();
   // Iterate over batches of input data
@@ -110,6 +113,7 @@ SimpleDMatrix::SimpleDMatrix(AdapterT* adapter, float missing, int nthread) {
     auto& batch = adapter->Value();
     auto batch_max_columns = sparse_page_.Push(batch, missing, nthread);
     inferred_num_columns = std::max(batch_max_columns, inferred_num_columns);
+    total_batch_size += batch.Size();
     // Append meta information if available
     if (batch.Labels() != nullptr) {
       auto& labels = info_.labels_.HostVector();
@@ -153,16 +157,27 @@ SimpleDMatrix::SimpleDMatrix(AdapterT* adapter, float missing, int nthread) {
     info_.num_col_ = adapter->NumColumns();
   }
 
+
   // Synchronise worker columns
   rabit::Allreduce<rabit::op::Max>(&info_.num_col_, 1);
 
   if (adapter->NumRows() == kAdapterUnknownSize) {
-    info_.num_row_ = offset_vec.size() - 1;
+    using IteratorAdapterT
+      = IteratorAdapter<DataIterHandle, XGBCallbackDataIterNext, XGBoostBatchCSR>;
+    if (std::is_same<AdapterT, IteratorAdapterT>::value
+        || std::is_same<AdapterT, FileAdapter>::value) {
+      info_.num_row_ = total_batch_size;
+      while (offset_vec.size() - 1 < total_batch_size) {
+        offset_vec.emplace_back(offset_vec.back());
+      }
+    } else {
+      CHECK((std::is_same<AdapterT, CSCAdapter>::value)) << "Expecting CSCAdapter";
+      info_.num_row_ = offset_vec.size() - 1;
+    }
   } else {
     if (offset_vec.empty()) {
       offset_vec.emplace_back(0);
     }
-
     while (offset_vec.size() - 1 < adapter->NumRows()) {
       offset_vec.emplace_back(offset_vec.back());
     }

--- a/tests/cpp/data/test_adapter.cc
+++ b/tests/cpp/data/test_adapter.cc
@@ -26,12 +26,13 @@ TEST(Adapter, CSRAdapter) {
   EXPECT_EQ(line0.GetElement(1).value, 2);
 
   auto line1 = batch.GetLine(1);
-  EXPECT_EQ(line1 .GetElement(0).value, 3);
-  EXPECT_EQ(line1 .GetElement(1).value, 4);
+  EXPECT_EQ(line1.GetElement(0).value, 3);
+  EXPECT_EQ(line1.GetElement(1).value, 4);
+
   auto line2 = batch.GetLine(2);
-  EXPECT_EQ(line2 .GetElement(0).value, 5);
-  EXPECT_EQ(line2 .GetElement(0).row_idx, 2);
-  EXPECT_EQ(line2 .GetElement(0).column_idx, 1);
+  EXPECT_EQ(line2.GetElement(0).value, 5);
+  EXPECT_EQ(line2.GetElement(0).row_idx, 2);
+  EXPECT_EQ(line2.GetElement(0).column_idx, 1);
 }
 
 TEST(Adapter, CSCAdapterColsMoreThanRows) {
@@ -73,10 +74,11 @@ class CSRIterForTest {
   std::vector<std::remove_pointer<decltype(std::declval<XGBoostBatchCSR>().index)>::type>
       feature_idx_ {0, 1, 0, 1, 1};
   std::vector<std::remove_pointer<decltype(std::declval<XGBoostBatchCSR>().offset)>::type>
-      row_ptr_ {0, 2, 4, 5};
+      row_ptr_ {0, 2, 4, 5, 5};
   size_t iter_ {0};
 
  public:
+  size_t static constexpr kRows { 4 };  // Test for the last row being empty
   size_t static constexpr kCols { 13 };  // Test for having some missing columns
 
   XGBoostBatchCSR Next() {
@@ -88,7 +90,7 @@ class CSRIterForTest {
     batch.offset = dmlc::BeginPtr(row_ptr_);
     batch.index = dmlc::BeginPtr(feature_idx_);
     batch.value = dmlc::BeginPtr(data_);
-    batch.size = 3;
+    batch.size = kRows;
 
     batch.label = nullptr;
     batch.weight = nullptr;
@@ -117,16 +119,23 @@ int CSRSetDataNextForTest(DataIterHandle data_handle,
   }
 }
 
-TEST(Adapter, IteratorAdaper) {
+TEST(Adapter, IteratorAdapter) {
   CSRIterForTest iter;
   data::IteratorAdapter<DataIterHandle, XGBCallbackDataIterNext,
                         XGBoostBatchCSR> adapter{&iter, CSRSetDataNextForTest};
-  constexpr size_t kRows { 6 };
+  constexpr size_t kRows { 8 };
 
   std::unique_ptr<DMatrix> data {
     DMatrix::Create(&adapter, std::numeric_limits<float>::quiet_NaN(), 1)
   };
   ASSERT_EQ(data->Info().num_col_, CSRIterForTest::kCols);
   ASSERT_EQ(data->Info().num_row_, kRows);
+  int num_batch = 0;
+  for (auto const& batch : data->GetBatches<SparsePage>()) {
+    ASSERT_EQ(batch.offset.HostVector(), std::vector<bst_row_t>({0, 2, 4, 5, 5, 7, 9, 10, 10}));
+    ++num_batch;
+  }
+  ASSERT_EQ(num_batch, 1);
 }
+
 }  // namespace xgboost

--- a/tests/cpp/data/test_simple_dmatrix.cc
+++ b/tests/cpp/data/test_simple_dmatrix.cc
@@ -190,12 +190,12 @@ TEST(SimpleDMatrix, FromFile) {
     std::ofstream fo(filename, std::ios::app | std::ios::out);
     fo << "0\n";
   }
-  constexpr size_t expected_nrow = 6;
+  constexpr size_t kExpectedNumRow = 6;
   std::unique_ptr<dmlc::Parser<uint32_t>> parser(
       dmlc::Parser<uint32_t>::Create(filename.c_str(), 0, 1, "auto"));
 
-  auto verify_batch = [](SparsePage const &batch) {
-    EXPECT_EQ(batch.Size(), expected_nrow);
+  auto verify_batch = [kExpectedNumRow](SparsePage const &batch) {
+    EXPECT_EQ(batch.Size(), kExpectedNumRow);
     EXPECT_EQ(batch.offset.HostVector(),
               std::vector<bst_row_t>({0, 3, 6, 9, 12, 15, 15}));
     EXPECT_EQ(batch.base_rowid, 0);

--- a/tests/cpp/data/test_simple_dmatrix.cc
+++ b/tests/cpp/data/test_simple_dmatrix.cc
@@ -185,16 +185,22 @@ TEST(SimpleDMatrix, FromCSC) {
 TEST(SimpleDMatrix, FromFile) {
   std::string filename = "test.libsvm";
   CreateBigTestData(filename, 3 * 5);
+  // Add an empty row at the end of the matrix
+  {
+    std::ofstream fo(filename, std::ios::app | std::ios::out);
+    fo << "0\n";
+  }
+  constexpr size_t expected_nrow = 6;
   std::unique_ptr<dmlc::Parser<uint32_t>> parser(
       dmlc::Parser<uint32_t>::Create(filename.c_str(), 0, 1, "auto"));
 
   auto verify_batch = [](SparsePage const &batch) {
-    EXPECT_EQ(batch.Size(), 5);
+    EXPECT_EQ(batch.Size(), expected_nrow);
     EXPECT_EQ(batch.offset.HostVector(),
-              std::vector<bst_row_t>({0, 3, 6, 9, 12, 15}));
+              std::vector<bst_row_t>({0, 3, 6, 9, 12, 15, 15}));
     EXPECT_EQ(batch.base_rowid, 0);
 
-    for (auto i = 0ull; i < batch.Size(); i++) {
+    for (auto i = 0ull; i < batch.Size() - 1; i++) {
       if (i % 2 == 0) {
         EXPECT_EQ(batch[i][0].index, 0);
         EXPECT_EQ(batch[i][1].index, 1);

